### PR TITLE
Build: Add gppkg v2 support for GP7

### PR DESCRIPTION
**Pull request by orhankislal**
- [ ] Add the module name, JIRA# to PR/commit and description.
- [ ] Add tests for the change.

### CODE REVIEW

File: src/ports/greenplum/cmake/FindGreenplum_6.cmake [hunk 1]
```diff
@@ -17,3 +17,64 @@
...
+    configure_file(
+        madlib.spec.in
+        \"\${CMAKE_CURRENT_BINARY_DIR}/${GPDB_VERSION}/SPECS/madlib.spec\"
+    )
+    configure_file(
+        gppkg_spec.yml.in
+        \"\${CMAKE_CURRENT_BINARY_DIR}/${GPDB_VERSION}/gppkg/gppkg_spec.yml\"
+    )
+
+    if(GPPKG_BINARY AND RPMBUILD_BINARY)
```
Comment (khannaekta) [inline src/ports/greenplum/cmake/FindGreenplum_6.cmake:54]: Would it be helpful to create a separate file `FindGreenplum_7.cmake`?

File: src/ports/postgres/modules/bayes/test/bayes.sql_in [hunk 2]
```diff
@@ -392,5 +392,17 @@ $$ language plpgsql;
...
-SELECT install_test_5();
+
+-- FIXME: ORCA started to fail with GP7 beta3 while accessing bayes created views in these tests
+m4_include(`SQLCommon.m4')
+m4_changequote(`<!', `!>')
+m4_ifdef(<!__POSTGRESQL__!>, <!!>, <!
+SET optimizer = off;
+!>)
+
+ SELECT install_test_4();
```
Comment (khannaekta) [inline src/ports/postgres/modules/bayes/test/bayes.sql_in:403]: do these test fail with gppkg v2 installation only or with GP7 beta3 in general ?

File: src/ports/greenplum/cmake/FindGreenplum_6.cmake (hunk 1 above)
Comment (orhankislal) [reply src/ports/greenplum/cmake/FindGreenplum_6.cmake:54]: This PR is for the madlib2-master so the focus is on GP7+ & PG13+. GP6 has py3 capabilities via an extension so it is deemphasised.

File: src/ports/postgres/modules/bayes/test/bayes.sql_in (hunk 2 above)
Comment (orhankislal) [reply src/ports/postgres/modules/bayes/test/bayes.sql_in:403]: I believe they fail with beta3. Even if we compile from source, we still fail these tests. I don't see any connection between ORCA optimizer failing to create a plan (even `explain` crashes the db) and gppkg.

File: src/ports/greenplum/cmake/FindGreenplum_6.cmake (hunk 1 above)
Comment (khannaekta) [reply src/ports/greenplum/cmake/FindGreenplum_6.cmake:54]: Thanks! Was just suggesting it as we have a separate `.cmake` for each version.

**Review by khannaekta — APPROVED** [review]
LGTM!

File: src/ports/greenplum/cmake/FindGreenplum_6.cmake (hunk 1 above)
Comment (orhankislal) [reply src/ports/greenplum/cmake/FindGreenplum_6.cmake:54]: We do have `FindGreenplum_7.cmake` in the same folder (in `madlib2-master` branch). It was added in a previous commit that added gp7 support.

Comment (khannaekta) [reply src/ports/greenplum/cmake/FindGreenplum_6.cmake:54]: Just realized, I was looking at the main branch.

**Pull request merged**